### DISCUSSION TITLE
Why do we need `throw null` in `src/libraries/System.Collections/ref` and that folder in general?

### DISCUSSION THREAD

**Discussion by B1Z0N** [dotnet/runtime#66701] [Q&A] [answered]
Why do we need that?

**Comment by EgorBo** [comment, marked as answer]
<library>/ref is auto-generated tiny facades. They are used in order to produce small-sized referenced assemblies needed for faster compilation - they're not used during the actual application run.

`throw null` is basically a universal way to implement a method body - it's never invoked in practice.

**Reply by B1Z0N** [reply]
Oh, I see, the `Libs.Ref` subset triggers `RefOnly` mode and it uses `ref` instead of `src`. Just two more questions:

1. What generates this `ref` directory?
2. In what case I'd want to use this RefOnly mode, what scenario?

Thanks a lot for your answer!

**Reply by bartonjs** [reply]
https://docs.microsoft.com/en-us/dotnet/standard/assembly/reference-assemblies

Two reasons that we use them:
* It provides a single file that shows us just the public API revision history.
* It helps us (with additional tooling) make sure that any library that builds with OS-specific flavors all share the same public API -- the public API that we defined in the reference assembly.
